cincinnati/src/plugins/internal/graph_builder/dkrv2_openshift_secondary_metadata_scraper/plugin.rs

```diff
@@ -0,0 +1,449 @@
+use crate as cincinnati;
+use crate::plugins::internal::release_scrape_dockerv2::registry;
+use std::path::{Path, PathBuf};
+use tempfile::TempDir;
+
+use self::cincinnati::plugins::prelude::*;
+use self::cincinnati::plugins::prelude_plugin_impl::*;
+
+use tokio::sync::Mutex as FuturesMutex;
+
+pub static DEFAULT_OUTPUT_WHITELIST: &[&str] = &[
+    "channels/.+\\.ya+ml",
+    "blocked-edges/.+\\.ya+ml",
+    "raw/metadata.json",
+    "version",
+];
+
+pub static DEFAULT_METADATA_IMAGE_REGISTRY: &str = "";
+pub static DEFAULT_METADATA_IMAGE_REPOSITORY: &str = "";
+pub static DEFAULT_METADATA_IMAGE_TAG: &str = "latest";
+
+// Defines the key for placing the data directory path in the IO parameters
+pub static GRAPH_DATA_DIR_PARAM_KEY: &str = "io.openshift.upgrades.secondary_metadata.directory";
+
+/// Plugin settings.
+#[derive(Debug, SmartDefault, Clone, Deserialize)]
+#[serde(default)]
+pub struct DkrV2OpenshiftSecondaryMetadataScraperSettings {
+    /// Directory where the image will be unpacked. Will be created if it doesn't exist.
+    output_directory: PathBuf,
+
+    /// Vector of regular expressions used as a positive output filter.
+    /// An empty vector is regarded as a configuration error.
+    #[default(DEFAULT_OUTPUT_WHITELIST.iter().map(|s| (*s).to_string()).collect())]
+    output_allowlist: Vec<String>,
```

Comment:
What is the role of output_allowlist?

Reply:
Only paths that match the regular expressions in it will be kept on the filesystem after image extraction.